R/aio.R

```diff
@@ -157,7 +164,58 @@ recv_aio_signal <- function(con,
                                      "integer", "logical", "numeric", "raw", "string"),
                             timeout = NULL,
                             n = 65536L)
-  data <- .Call(rnng_recv_aio_signal, con, cv, mode, timeout, n, environment())
+  data <- .Call(rnng_recv_aio_signal, con, cv, mode, timeout, n, environment(),
+    function() {
+      cb <- data$callback
+      if (!is.null(cb)) {
+        cb(data)
+      }
+    }
+  )
+
+#' @exportS3Method promises::is.promising
+is.promising.recvAio <- function(x) {
+  TRUE
+}
+
+#' @exportS3Method promises::as.promise
+as.promise.recvAio <- function(x) {
+  prom <- x$promise
+
+  if (is.null(prom)) {
+    prom <- promises::promise(function(resolve, reject) {
+      assign("callback", function(...) {
```

Comment:
Had to use `assign` here because `$<-.recvAio` is a no-op. (Took me quite some time to realize that! 😅)